src/pinnable_mapped_file.cpp

```diff
@@ -97,22 +97,22 @@ pinnable_mapped_file::pinnable_mapped_file(const bfs::path& dir, bool writable,
    }
 
    segment_manager* file_mapped_segment_manager = nullptr;
-   if(!bfs::exists(_data_file_path)) {
+   if(!std::filesystem::exists(_data_file_path)) {
       std::ofstream ofs(_data_file_path.generic_string(), std::ofstream::trunc);
-      //win32 impl of bfs::resize_file() doesn't like the file being open
+      //win32 impl of std::filesystem::resize_file() doesn't like the file being open
```

Comment:
Do we know if this comment is true? This was fixed in boost 1.79. I suspect it is not true of `std::filesystem`. The cppreference says it does POSIX `truncate` which says "The truncate() function shall not modify the file offset for any open file descriptions associated with the file."

Reply:
I am not sure if it's true or not for win32 right now. However, I don't think close the file here have any performance implication. I'll just keep the implementation as it was and remove the comment for now.